spec.html

```diff
@@ -12014,8 +12014,8 @@ <h1>Agents</h1>
         </tr>
         <tr>
           <td>[[KeptAlive]]</td>
-          <td>a List of Objects</td>
-          <td>Initially a new empty List, representing the list of objects to be kept alive until the end of the current Job</td>
+          <td>a List of either Objects or Symbols</td>
```

Comment:
A conversation with ChatGPT:


> Me: In the phrase: "A bag of either apples or oranges" does this mean that the bag can contain either only applies or only oranges? Or does it mean that the bag can contain a mix of both apples and oragnes?
>
> AI: The phrase "A bag of either apples or oranges" means that the bag can contain either only apples or only oranges, not a mix of both.
>
> Me: are you sure?
>
> AI: It is possible that the phrase "A bag of either apples or oranges" is being used to indicate that the bag could contain a mix of both apples and oranges, however, without more context it is impossible to know for certain. The phrase "A bag of either apples or oranges" is often used to indicate that the bag can only contain one type of fruit, either apples or oranges.

Would any of the following be clearer?

 - `a List of ECMAScript Language values`
 - `a List of Objects and/or Symbols`
 - `a List of Objects and Symbols`
 
 CC: @jmdyck

Reply:
> > The phrase "A bag of either apples or oranges" is often used to indicate that the bag can only contain one type of fruit, either apples or oranges.

Yeah, that's probably fair. 
 
> Would any of the following be clearer?
> 
>     * `a List of ECMAScript Language values`

Correct, but imprecise.
 
>     * `a List of Objects and/or Symbols`

It's a possibility.

>     * `a List of Objects and Symbols`

In the apples+oranges analogy, "a bag of apples and oranges" suggests it has some of each kind of fruit. So it wouldn't be clear that (e.g.) a List of only objects would satisfy the description.

---
The status quo isn't entirely consistent on how to describe such lists:

4 times `a List of either X or Y`:
- 16.2.1.6.2 (return type): `a List of either Strings or *null*`
- 22.2.7.8 (type of _indices_): `a List of either Match Records or *undefined*`
- 22.2.7.8 (type of _groupNames_): `a List of either Strings or *undefined*`
- 29.5.4 (type of _Ws_): `a List of either WriteSharedMemory or ReadModifyWriteSharedMemory events`

1 time `a List of X or Y` (but see below re ambiguity):
- 29.5.5 (assertion on _Ws_): `a List of WriteSharedMemory or ReadModifyWriteSharedMemory events ...`

2 times `a List ... whose elements ...`
- 22.1.3.18.1 (type of _captures_): `a possibly empty List, each of whose elements is a String or *undefined*`
- 29.6.2 (condition): `a List ... whose elements are WriteSharedMemory or ReadModifyWriteSharedMemory events ...`

2 times break it out as a separate sentence (which isn't much of an option in the above contexts):
- 6.1.7.3 ([[OwnPropertyKeys]]): `The Type of each element of the returned List is either String or Symbol.`
- 10.5.11 (Note re [[OwnPropertyKeys]]): `The Type of each result List element is either String or Symbol.`

So my advice is to use `a List of either Objects or Symbols` for this PR, and then raise an issue/PR to establish an editorial convention on this question.

----

Note that, in general, `a List of X or Y` can be ambiguous.
E.g., `a List of Strings or *undefined*` could mean:
- `a List of (Strings or *undefined*)`
or
- `(a List of Strings) or *undefined*`

The spec uses 'either' to distinguish these two meanings:
- `a List of either Strings or *undefined*` (the type of _groupNames_ noted above)
vs
- `either a List of Strings or *undefined*` (the type of [[PropertyList]] in 25.5.2.1)